lib/screenplay/suppressed_predictions/suppressed_prediction.ex

```diff
@@ -0,0 +1,150 @@
+    )
+    |> validate_edge_cases()
+    |> validate_location(places)
+    |> validate_routes(places)
+  end
+
+  defp validate_edge_cases(changeset) do
+    location_id = get_field(changeset, :location_id)
+    route_id = get_field(changeset, :route_id)
+
+    cond do
+      # JFK/UMass has the Ashmont/Braintree split, we ask for those stop ids rather than the parent id
+      location_id == "place-jfk" ->
+        add_error(
+          changeset,
+          :location_id,
+          "Please provide either the Ashmont or Braintree platform location_id: (#{@jfk_umass_ashmont_place} or #{@jfk_umass_braintree_place})"
+        )
+
+      # We don't handle separate Green Line and Silver Line branching route_ids
+      # That is populated when we pass data to Transit Data
+      # For internal use we just use "Green" or "Silver" to show all green/silver line routes
+      route_id in @green_line_routes ->
+        add_error(
+          changeset,
+          :route_id,
+          "Please provide just `Green` as the route_id for handling all Green Line routes"
+        )
+
+      true ->
+        changeset
+    end
+  end
+
+  defp validate_location(changeset, places) do
+    location_id = get_field(changeset, :location_id)
+
+    if places |> Enum.any?(&(&1.id == location_id)) do
+      changeset
+    else
+      add_error(changeset, :location_id, "Location `#{location_id}` does not exist")
+    end
+  end
+
+  defp validate_routes(changeset, places) do
+    unchecked_location_id = get_field(changeset, :location_id)
+    route_id = get_field(changeset, :route_id)
+
+    location_id =
+      if unchecked_location_id in [@jfk_umass_ashmont_place, @jfk_umass_braintree_place] do
+        "place-jfk"
+      else
+        unchecked_location_id
+      end
+
+    places
+    |> Enum.find(fn place -> place.id == location_id end)
+    |> case do
+      nil ->
+        add_error(
+          changeset,
+          :route_id,
+          "Location `#{location_id}` does not exist"
+        )
+
+      place ->
+        case route_id do
+          "Silver" ->
+            # For Silver Line, we make sure that the routes are all valid routes in Screenplay
+            # In this case, 741, 742, 743 and 746 within @valid_silver_line_routes
+            if Enum.any?(
+                 place.screens,
+                 fn
+                   %Screenplay.Places.Place.PaEssScreen{routes: routes} ->
+                     Enum.any?(
+                       routes,
+                       fn route -> route.id in @valid_silver_line_routes end
+                     )
+
+                   _ ->
+                     false
+                 end
+               ) do
+              changeset
+            else
+              add_error(
+                changeset,
+                :route_id,
+                "Place `#{place.id}` currently has only Silver Line routes that are not supported in Screenplay"
+              )
+            end
+
+          _ ->
+            changeset
```

Comment:
The silver case looks well handled, but this is a bit too permissive. If we add a case for Green that checks if any of the place's routes starts with "Green", and a default case that checks if the route_id is in the place's routes, then we should be good.